static_frame/core/index_hierarchy.py

```diff
@@ -1711,29 +1712,22 @@ def _loc_to_iloc_index_hierarchy(self: IH,
         if key._recache:
             key._update_array_cache()
 
-        key_indexers: tp.List[np.ndaray] = []
+        remapped_indexers: tp.List[np.ndaray] = []
         for key_index, self_index, key_indexer in zip(
                 key._indices,
                 self._indices,
                 key._indexers,
                 ):
             indexer_remap = key_index._index_iloc_map(self_index)
-            key_indexers.append(indexer_remap[key_indexer])
-
-        key_indexers = np.array(key_indexers, dtype=DTYPE_INT_DEFAULT).T
+            remapped_indexers.append(indexer_remap[key_indexer])
 
-        _, ilocs, ilocs_order = np.intersect1d(
-                view_2d_as_1d(self._indexers.T),
-                view_2d_as_1d(key_indexers),
-                return_indices=True,
-                )
+        remapped_indexers = np.array(remapped_indexers, dtype=DTYPE_UINT_DEFAULT).T
 
-        if len(ilocs) != len(key):
-            raise KeyError(
-                f"Key contains labels that are not in this IndexHierarchy, including: {key._drop_iloc(ilocs)[0]}"
-            )
-
-        return ilocs[ilocs_order]
+        try:
```

Comment:
The tricky part of this function, is that it's not enough to remap.

For example, lets say `A` is:
```
0, 0
1, 1
```
and `B` is
```
1, 0
0, 1
```
The remapping would work, since both indexers share the same labels at each depth.

The problem is, that `A` doesn't contain the label `(1, 0)`. I originally addressed this by using `intersect1d`, but that led to the issue where order wasn't being retained. In the last fix attempt, I incorrectly assumed from one example that doing `ilocs[ilocs_order]` was enough to ensure the order was retained, but that was not correct.

I realized that the better approach would be to take advantage of `self._map`, which already has optimized infrastructure to map indexers to unique integer locations. So I could simply take the `remapped_indexers`, and then map them to iloc locations. If any of them are invalid, then the mapping will raise a `KeyError`.